pandas/io/parsers.py

```diff
@@ -3422,10 +3435,12 @@ def get_rows(self, n, skiprows=None):
         self.buffer = iter(buffer_rows)
         return detect_rows
 
-    def detect_colspecs(self, n=100, skiprows=None):
+    def detect_colspecs(self, n=100, skiprows=None, infer_nrows=None):
```

Comment:
That's what I had proposed initially and had replaced `n` with `infer_nrows` in place. Your comment about putting the new keyword as a trailing argument is what made me realize that it was possible for someone's code to have instantiated FixedWidthReader and explicitly set `n`. 

Granted, that seems like a rare use case, but I was able to implement a solution that handled both cases. However, this solution seems to have come at the cost of confusing documentation, where `infer_nrows` has a default of ``None`` in order for `read_fwf(colspecs='infer')` to default to inferring from the first 100 rows, as was the case beforehand.

If you and @jreback prefer, I'm happy to use either
`detect_colspecs(self, infer_nrows=100, skiprows=None)` or
`detect_colspecs(self, skiprows=None, infer_nrows=100)`
with the awareness that it might be a breaking change in rare cases.